### PR TITLE
Added transport feedbacks and fixed connection issues

### LINKED ISSUE #4: Disconnections not handled properly
When a device disconnects (in our case, weekly reboots), there are a couple issues:

- Companion doesn't catch it sometimes; other times it sees it (`Network error: read ECONNRESET`)
- Status doesn't always change
- Reconnections, while it may happen, doesn't appear to work properly after a disconnection
- Disabling and then reenabling reconnects it

My thoughts are there's something wrong in reconnecting. I'll be working on a fix, I just wanted to document what the problem was.

A work around we have right now is to reconnect to the device before sending a command. Obviously that's a hacky solution, but is a work around.

### PR DESCRIPTION
**Pull request by onfire4g05**
This fixes several bugs and adds the channel status variable. It also transitions the code to ES6.

Tested on a 2 channel GV K2 Summit.

## Adds
- Transport ~variable~ feedbacks to check the channel status (playing, stopped, etc)
- ES6

## Fixes
- Connection issues (reconnections would not work if it was waiting for a response, reconnections aren't always noticed since there was no timeout with constant poll/ping) (Fixes #4)
- Issue where the buffer property could grow to an infinite size since it was always appended to
- Issue where clip names won't always populate correctly

### CODE REVIEW

**Comment by onfire4g05** [comment]
Swapped to feedbacks for status vs variables. The value is also now cached, so that logic isn't run every status update.

File: index.js [hunk 1]
```diff
@@ -22,394 +23,483 @@ function str2hex(str) {
...
+		};
 
-	self.awaiting_reply = false;
-	self.command_queue = [];
-	self.files = [];
+		this.actions(); // export actions
 
-	// super-constructor
-	instance_skel.apply(this, arguments);
+		return this;
```
Comment (krocheck) [inline index.js]: `return this;` shouldn't be necessary in the constructor, though I don't believe there's a penalty for it either.

File: index.js [hunk 2]
```diff
@@ -22,394 +23,483 @@ function str2hex(str) {
...
+	}
 
-	self.actions(); // export actions
+	updateConfig(config) {
+		this.config = config;
+		this.init_tcp();
```
Comment (krocheck) [inline index.js]: I typically put detection in here to test if the IP changed and cycle the connection only if true, but this is technically valid and what the module was doing prior.

File: index.js [hunk 3]
```diff
@@ -22,394 +23,483 @@ function str2hex(str) {
...
+		this.initFeedbacks();
+	}
 
-instance.prototype.updateConfig = function(config) {
-	var self = this;
+	init_tcp() {
+		if (this.socket !== undefined) {
+			this.socket.destroy();
+			delete this.socket;
```
Comment (krocheck) [inline index.js]: Shouldn't the transport timer be checked and cleared if present as well?

**Comment by krocheck** [comment]
@onfire4g05 did you see my inline notes?

**Comment by onfire4g05** [comment]
Yes. I'm on vacation this week with somewhat limited Internet. I can fix the issues next week!

**Comment by onfire4g05** [comment]
This will also now allow for clip names (record or load) to use companion variables.

**Pull request merged**